third-party/spamassassin/src/test/java/org/apache/james/spamassassin/mock/MockSpamdExtension.java

```diff
@@ -23,25 +23,29 @@
 import java.util.concurrent.Executors;
 
 import org.apache.james.util.concurrent.NamedThreadFactory;
+import org.junit.jupiter.api.extension.AfterEachCallback;
+import org.junit.jupiter.api.extension.BeforeAllCallback;
+import org.junit.jupiter.api.extension.BeforeEachCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
 import org.junit.rules.ExternalResource;
 
-public class MockSpamdTestRule extends ExternalResource {
+public class MockSpamdExtension implements AfterEachCallback, BeforeEachCallback {
```

Comment:
the last of the public modifiers

I know how painful it is to do this, I won't blame you if you don't want to do it :)

Reply:
Not this one, it's an extension declaration that will be used in oither Test classes `SpamAssassinTest` and `SpamAssassinHandlerTest` :)